mflowgen/tile_array/custom-init/outputs/floorplan.tcl

```diff
@@ -0,0 +1,160 @@
+#=========================================================================
+# floorplan.tcl
+#=========================================================================
+# This script is called from the Innovus init flow step.
+#
+# Author : Christopher Torng
+# Date   : March 26, 2018
+
+#-------------------------------------------------------------------------
+# Floorplan variables
+#-------------------------------------------------------------------------
+
+set tile_separation_x 0
+set tile_separation_y 0
+set grid_margin_t 50
+set grid_margin_b 50
+set grid_margin_l 200
+set grid_margin_r 200
+
+# Make room in the floorplan for the core power ring
+
+set pwr_net_list {VDD VSS}; # List of power nets in the core power ring
+
+set M1_min_width   [dbGet [dbGetLayerByZ 1].minWidth]
+set M1_min_spacing [dbGet [dbGetLayerByZ 1].minSpacing]
+
+set savedvars(p_ring_width)   [expr 48 * $M1_min_width];   # Arbitrary!
+set savedvars(p_ring_spacing) [expr 24 * $M1_min_spacing]; # Arbitrary!
+
+# Core bounding box margins
+
+set core_margin_t [expr ([llength $pwr_net_list] * ($savedvars(p_ring_width) + $savedvars(p_ring_spacing))) + $savedvars(p_ring_spacing)]
+set core_margin_b [expr ([llength $pwr_net_list] * ($savedvars(p_ring_width) + $savedvars(p_ring_spacing))) + $savedvars(p_ring_spacing)]
+set core_margin_r [expr ([llength $pwr_net_list] * ($savedvars(p_ring_width) + $savedvars(p_ring_spacing))) + $savedvars(p_ring_spacing)]
+set core_margin_l [expr ([llength $pwr_net_list] * ($savedvars(p_ring_width) + $savedvars(p_ring_spacing))) + $savedvars(p_ring_spacing)]
```

Comment:
Do we want these since we don't have a power ring here? I realize I need to remove them from the glb stuff as well...

Reply:
Yeah, we should probably remove, or at least change the way this is done so that the variables don't reference any non-existent power rings. Not crucial now, thought. We should talk about how to change how we add margins to all our blocks, and perhaps discuss some way of reducing the about of duplicated code we have across our blocks for this.